experimental/packages/opentelemetry-sdk-metrics/src/export/MetricExporter.ts

```diff
@@ -40,12 +38,6 @@ export interface PushMetricExporter {
    */
   forceFlush(): Promise<void>;
 
-  /**
-   * Select the {@link AggregationTemporality} for the given
-   * {@link InstrumentType} for this exporter.
-   */
-  selectAggregationTemporality(instrumentType: InstrumentType): AggregationTemporality;
```

Comment:
Reading the spec, I think it is still allowed to have the `PushMetricExporter` provide an `AggregationTemporalityPreference`. From what I have seen, allowing automatic configuration of the MetricReader via the Exporter can reduce setup code for third-party exporters quite a bit and make the setup less error prone at the same time. 

I'm wondering if we should keep it in some form. :thinking:

Reply:
> The `aggregation` and `temporality` properties used by the OpenTelemetry Metric SDK are determined when registering Metric Exporters through their associated MetricReader.
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricexporter

I'd read this statement that the `aggregation` and `temporality` still need to be configured with MetricReader. When we add automatic metric reader/exporter configuration with environs, we can configure the MetricReader with the preferred aggregation and temporality.

Reply:
Hmm, interesting. :thinking: Could be that I'm misunderstanding the spec. I interpreted that paragraph to mean that we can implement a mechanism to allow exporters to provide defaults for `aggregation` and `temporality` that may be overridden by the reader's explicit config. This way the reader would pull configuration from the exporter unless it has an explicit config.

Reply:
> The aggregation and temporality properties used by the OpenTelemetry Metric SDK are determined when registering Metric Exporters through their associated [MetricReader](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader).

I just take this to mean it is determined at this time, and cannot be manipulated later. The purpose of this is to ensure an exporter always receives the same temporality for a metric.

> OpenTelemetry language implementations MAY support automatically configuring the MetricReader to use for an Exporter.

This is quite broad, but I think it allows the scenario where if a user hasn't explicitly configured a temporality with the metric reader, it can get a temporality preference from the exporter. This is the way the other SDKs have also implemented it.

Reply:
Thanks for sharing the ideas. I've submitted https://github.com/open-telemetry/opentelemetry-specification/issues/2746 for clarification.

Reply:
I'll defer this change to the resolution of https://github.com/open-telemetry/opentelemetry-specification/issues/2746. Changes reverted.

Reply:
Thinking about it, we can probably make the aggregation temporality selector an optional property on the `PushMetricExporter`, read it in the `MetricReader` and default back to all-cumulative. This way it would be no duplicated work for people that use cumulative only but would still allow vendors to specify something different if that is necessary for their backends. But that is something for another PR :)

Reply:
Created a new issue: https://github.com/open-telemetry/opentelemetry-js/issues/3213
I'll see how much work it would be, but I can't promise to pick it up immediately. :slightly_smiling_face: